api/query.go

```diff
@@ -57,32 +57,25 @@ func (e *Endpoint) slice(ctx *gin.Context, query SliceQuery) {
 	writeResponse(ctx, metadata, data)
 }
 
-func sliceParseGetReq(ctx *gin.Context) (*SliceQuery, error) {
-	var query SliceQuery
-
-	err := json.Unmarshal([]byte(ctx.Query("query")), &query)
-	if err != nil {
-		return nil, err
-	}
-
-	if err = binding.Validator.ValidateStruct(&query); err != nil {
-		return nil, err
+func parseGetRequest(ctx *gin.Context, v interface{}) error {
+	if err := json.Unmarshal([]byte(ctx.Query("query")), v); err != nil {
+		return err
 	}
 
-	return &query, nil
+	return binding.Validator.ValidateStruct(v);
```

Comment:
We also might want to discuss and agree what we want to do with formatters.
Default golang formatter (no idea which one it is), allows formatting only the full document, not the selection. So that means that if use formatter later on unformatted file, many lines will appear changed and it might create difficulties on review.

In this case btw it removes semicolon :D